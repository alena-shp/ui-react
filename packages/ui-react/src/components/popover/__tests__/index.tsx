import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';

import { Popover } from '..';

test('Default use', async () => {
    const { asFragment, getByText } = render(
        <Popover
            children="Text trigger"
            render="Text render"
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Arrow', async () => {
    const { asFragment, getByText } = render(
        <Popover
            children="Text trigger"
            render="Text render"
            arrow={true}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('CustomTagName', async () => {
    const { asFragment, getByText } = render(
        <Popover
            children="Text trigger"
            render="Text render"
            triggerTagName="section"
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Close', async () => {
    const { asFragment, getByText } = render(
        <div>
            <Popover
                children="Text trigger"
                render="Text render"
            />
            <div>Out click</div>
        </div>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });

    fireEvent.click(getByText('Out click'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Element', async () => {
    const { asFragment, getByText } = render(
        <Popover
            children={<a href="#">Text trigger</a>}
            render="Text render"
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Hover', async () => {
    const { asFragment, getByText } = render(
        <Popover
            children="Text trigger"
            render="Text render"
            trigger="hover"
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Render function', async () => {
    const { asFragment, getByText } = render(
        <Popover
            children="Text trigger"
            render={() => 'Text render'}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Text trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});
