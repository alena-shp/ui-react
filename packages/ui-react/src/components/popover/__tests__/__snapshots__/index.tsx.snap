// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow 1`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
  </div>
</DocumentFragment>
`;

exports[`Arrow 2`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
    <div
      class="popover__content"
      style="position: fixed; top: 0px; left: 0px;"
    >
      Text render
      <svg
        height="15.999999999999998"
        preserveAspectRatio="xMaxYMax"
        style="position: absolute; top: 0px; left: 0px; transform: translateX(-50%);"
        viewBox="0 0 15.999999999999998 8"
        width="15.999999999999998"
      >
        <path
          d="M 0 8 H 15.999999999999998 C 8.999999999999998 1 7.999999999999999 0 7.999999999999999 0 C 7.999999999999999 0 6.999999999999999 1 0 8"
          fill="white"
          stroke="black"
          stroke-width="0"
        />
        <path
          d="M 0 8 H 15.999999999999998 L 15.999999999999998 8 H 0 Z"
          fill="white"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Close 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="popover__trigger"
      >
        Text trigger
      </div>
    </div>
    <div>
      Out click
    </div>
  </div>
</DocumentFragment>
`;

exports[`Close 2`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="popover__trigger"
      >
        Text trigger
      </div>
      <div
        class="popover__content"
        style="position: fixed; top: 0px; left: 0px;"
      >
        Text render
      </div>
    </div>
    <div>
      Out click
    </div>
  </div>
</DocumentFragment>
`;

exports[`Close 3`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="popover__trigger"
      >
        Text trigger
      </div>
    </div>
    <div>
      Out click
    </div>
  </div>
</DocumentFragment>
`;

exports[`Default use 1`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
  </div>
</DocumentFragment>
`;

exports[`Default use 2`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
    <div
      class="popover__content"
      style="position: fixed; top: 0px; left: 0px;"
    >
      Text render
    </div>
  </div>
</DocumentFragment>
`;

exports[`Element 1`] = `
<DocumentFragment>
  <div>
    <a
      href="#"
    >
      Text trigger
    </a>
  </div>
</DocumentFragment>
`;

exports[`Element 2`] = `
<DocumentFragment>
  <div>
    <a
      href="#"
    >
      Text trigger
    </a>
    <div
      class="popover__content"
      style="position: fixed; top: 0px; left: 0px;"
    >
      Text render
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hover 1`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hover 2`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render function 1`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render function 2`] = `
<DocumentFragment>
  <div>
    <div
      class="popover__trigger"
    >
      Text trigger
    </div>
    <div
      class="popover__content"
      style="position: fixed; top: 0px; left: 0px;"
    >
      Text render
    </div>
  </div>
</DocumentFragment>
`;
